### PR TITLE
fix(issue#607): Added personalizaton for reports - added field addedBy into CourseReport model.

### PR DESCRIPTION
**Pull request by alexbojko**
## [SOLUTION] Add field addedBy into CourseReport model to know who created this report.

Each report will be stored in folder with name as user id who created this report.
Need to run migrations.

### CODE REVIEW

File: mysite/analytics/models.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
 from django.db import models
+from django.contrib.auth.models import User
 from filer.fields.file import FilerFileField
 
 from ct.models import Course
 
+def get_upload_function(folder=''):
+    def user_dir_path(instance, filename):
+        return '{}/{}/{}'.format(folder, instance.addedBy.id, filename)
```
Comment (rimikt) [inline mysite/analytics/models.py]: Please use `os.path.join` for this.

File: mysite/api/v0/views.py [hunk 2]
```diff
@@ -55,13 +55,15 @@ class GenReportView(APIView):
     authentication_classes = (SessionAuthentication,)
 
     def get(self, request, format=None):
+        print request.user
```
Comment (rimikt) [inline mysite/api/v0/views.py]: Remove print statement

File: mysite/templates/ct/edit_course.html [hunk 3]
```diff
@@ -53,6 +53,13 @@
         <li>
             <a href='{% static report.response_report.url %}'>Download</a>
             report from {{ report.date }}
+            {% if report.addedBy %} created by
```
Comment (rimikt) [inline mysite/templates/ct/edit_course.html]: Please render only personalized reports - created by current user.

File: mysite/analytics/models.py [hunk 4]
```diff
@@ -1,14 +1,21 @@
 from django.db import models
+from django.contrib.auth.models import User
 from filer.fields.file import FilerFileField
 
 from ct.models import Course
 
+def get_upload_function(folder=''):
```
Comment (rimikt) [inline mysite/analytics/models.py]: Add docstring where specify how we generate this path. Thanks.

File: mysite/templates/ct/edit_course.html (hunk 3 above)
Comment (alexbojko) [inline mysite/templates/ct/edit_course.html]: we decided to leave it as it is

File: mysite/api/v0/views.py [hunk 5]
```diff
@@ -59,9 +59,10 @@ def get(self, request, format=None):
         if not course_id:
             return RestResponse('course_id is not provided', status=400)
         course = get_object_or_404(Course, id=course_id)
-        if not course.addedBy == request.user:
+        course_instructors = course.role_set.filter(role=Role.INSTRUCTOR).values_list('user_id', flat=True)
+        if request.user.id not in course_instructors and not course.addedBy == request.user:
```
Comment (rimikt) [inline mysite/api/v0/views.py]: This diff makes course_instructors not usable.

**Pull request merged**